src/Database/Driver/Mysql.php

```diff
@@ -230,6 +231,32 @@ public function supportsDynamicConstraints(): bool
         return true;
     }
 
+    /**
+     * @inheritDoc
+     */
+    protected function _expressionTranslators(): array
+    {
+        $namespace = 'Cake\Database\Expression';
+
+        return [
+            $namespace . '\WithExpression' => '_transformWithExpression',
+        ];
+    }
+
+    /**
+     * Receives a WithExpression and changes it so that it conforms to this
+     * SQL dialect.
+     *
+     * @param \Cake\Database\Expression\WithExpression $expression The expression to transform.
+     * @return void
+     */
+    protected function _transformWithExpression(WithExpression $expression): void
+    {
+        foreach ($expression->getExpressions() as $expression) {
+            $expression->setModifiers([]);
```

Comment:
Should we throw an exception here? I don't think it's normal cake convention to hide unsupported sql, is it?

Reply:
I'm not sure, I guess most transformers do usually transform into an equivalent. Gotta have a look at it.

Reply:
Unless there is a different sql syntax, I prefer that we simply generate the sql regardless. Users will receive a sql error if they use unsupported features.

Reply:
Fair enough. It can be revisited in case other dialects implement equivalent modifiers

Reply:
done